File: easy/py/easy-power-of-thor-episode-1.py

# EASY: https://www.codingame.com/training/easy/power-of-thor-episode-1
import sys
import math

# Hint: You can use the debug stream to print initialTX and initialTY, if Thor seems not follow your orders.

# light_x: the X position of the light of power
# light_y: the Y position of the light of power
# initial_tx: Thor's starting X position
# initial_ty: Thor's starting Y position
light_x, light_y, initial_tx, initial_ty = [int(i) for i in input().split()]

# game loop
while True:
    remaining_turns = int(input())  # The remaining amount of turns Thor can move. Do not remove this line.
   
    moveWE = light_x - initial_tx
    moveSN = initial_ty - light_y

    initial_tx += 1 if moveSN == 0 or moveWE > 0 else - 1 if moveWE < 0 else 0
    initial_ty += 1 if moveWE == 0 or moveSN < 0 else - 1 if moveSN > 0 else 0

    print(("N" if moveSN > 0 else 'S' if moveSN < 0 else '') + ('E' if moveWE > 0 else 'W' if moveWE < 0 else ''))

File: easy/js/easy-power-of-thor-episode-1.js
//EASY: https://www.codingame.com/training/easy/power-of-thor-episode-1

var inputs = readline().split(" ");
const lightX = parseInt(inputs[0]); // the X position of the light of power
const lightY = parseInt(inputs[1]); // the Y position of the light of power
let initialTx = parseInt(inputs[2]); // Thor's starting X position
let initialTy = parseInt(inputs[3]); // Thor's starting Y position

// game loop
while (true) {
  const remainingTurns = parseInt(readline()); // The remaining amount of turns Thor can move. Do not remove this line.

  let moveWE = lightX - initialTx;
  let moveSN = initialTy - lightY;

  if (moveSN === 0) initialTx += 1;
  else if (moveWE === 0) initialTy += 1;
  else (initialTx += moveWE > 0 ? -1 : 1), (initialTy += moveSN > 0 ? -1 : 1);

  console.log(
    (moveSN > 0 ? "N" : moveSN < 0 ? "S" : "") +
      (moveWE > 0 ? "E" : moveWE < 0 ? "W" : "")
  );
}
